substrate/in-instructions/primitives/src/shorthand.rs

```diff
@@ -0,0 +1,42 @@
+use scale::{Encode, Decode, MaxEncodedLen};
+use scale_info::TypeInfo;
+
+#[cfg(feature = "std")]
+use serde::{Serialize, Deserialize};
+
+use sp_core::{ConstU32, bounded::BoundedVec};
+
+use serai_primitives::{NativeAddress, Coin, Amount};
+
+use crate::{MAX_DATA_LEN, ExternalAddress, ExternalInInstruction, OutInstruction};
+
+#[derive(Clone, PartialEq, Eq, Debug, Encode, Decode, MaxEncodedLen, TypeInfo)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum Shorthand {
+  Raw(BoundedVec<u8, ConstU32<{ MAX_DATA_LEN }>>),
+  Swap {
+    origin: Option<ExternalAddress>,
+    coin: Coin,
+    minimum: Amount,
+    out: OutInstruction,
+  },
+  AddLiquidity {
+    origin: Option<ExternalAddress>,
+    minimum: Amount,
+    gas: Amount,
```

Comment:
why gas for add liq?

Reply:
It's the gas turned over to the Serai account receiving the LP tokens. It lets you add XMR:SRI liq from Monero, and when the time comes to remove it, not need to do a Monero TX to acquire SRI for gas. This is documented in the docs.

Reply:
This is an unnecessary optimization in my opinion.

Reply:
This could be dropped for Mint alone and then the Serai call.